planner/implementation/datasource.go

```diff
@@ -108,3 +108,66 @@ func (impl *TableScanImpl) CalcCost(outCount float64, children ...memo.Implement
 	}
 	return impl.cost
 }
+
+// IndexReaderImpl is the implementation of PhysicalIndexReader.
+type IndexReaderImpl struct {
+	baseImpl
+	tblColHists *statistics.HistColl
+}
+
+// ScaleCostLimit implements Implementation interface.
+func (impl *IndexReaderImpl) ScaleCostLimit(costLimit float64) float64 {
+	reader := impl.plan.(*plannercore.PhysicalIndexReader)
+	sessVars := reader.SCtx().GetSessionVars()
+	copIterWorkers := float64(sessVars.DistSQLScanConcurrency)
+	if math.MaxFloat64/copIterWorkers < costLimit {
+		return costLimit
+	}
+	return costLimit * copIterWorkers
+}
+
+// CalcCost implements Implementation interface.
+func (impl *IndexReaderImpl) CalcCost(outCount float64, children ...memo.Implementation) float64 {
+	reader := impl.plan.(*plannercore.PhysicalIndexReader)
+	sessVars := reader.SCtx().GetSessionVars()
+	networkCost := outCount * sessVars.NetworkFactor * impl.tblColHists.GetAvgRowSize(children[0].GetPlan().Schema().Columns, true)
```

Comment:
Since we are using Arrow format in network transferring, the `isEncodedKey` param for `GetAvgRowSize` here may have problem.

Reply:
Do you mean the change in https://github.com/pingcap/tidb/pull/13461?

Reply:
Yes.